python/ray/ml/train/data_parallel_trainer.py

```diff
@@ -292,27 +292,36 @@ def training_loop(self) -> None:
         else:
             resume_checkpoint_dict = None
 
-        # Tell Ray Train to only shard the train dataset and not the other datasets.
-        # This is purely an implementation detail and users do not need to know about
-        # this.
-        # TODO(amog): Refactor this to remove hack and make this more modular.
-        #  TrainingIterator should accept a generic custom_ingest_func that contains
-        #  the logic for how to split the Datasets.
-        updated_dataset_dict = {}
-        for key, value in self.datasets.items():
-            if key == TRAIN_DATASET_KEY:
-                updated_dataset_dict[key] = value
-            else:
-                # Ray Train will strip out the added string before exposing to users.
-                updated_dataset_dict[key + "_NO-SHARD"] = value
+        def dataset_split_fn(dataset_dict, training_worker_handles):
```

Comment:
Could we pull this out into a default splitting function in the util module?

Reply:
If we're pulling this out into a default splitting function, could you add a docstring? Would allow readers to understand the function without having to reference `_RayDatasetSpec`.

Reply:
Separated into its own function, but left it in `data_parallel_trainer` for now as it is only being used in `DataParallelTrainer`. Let's revisit the location if we need it in more trainers in the future.